api/env.docker

```diff
@@ -5,8 +5,8 @@ DJANGO_SECRET_KEY="ny#b__$$f6ry4wy8oxre97&-68u_0lk3gw(z=d40_dxey3zw0v1"
 DJANGO_DEBUG_ENABLED=True
 
 BASE_URL=http://localhost:50280/
-ENVIRONMENT=development
-ALLOWED_HOSTS=api.openverse.engineering,api-dev.openverse.engineering,host.docker.internal
+ENVIRONMENT=local
```

Comment:
We use `local` elsewhere and it's a bit clearer than `development`, which has been historically muddled with `staging`.